types/api/order.d.ts

```diff
@@ -0,0 +1,187 @@
+import { Part } from "../base";
+
+export type OrderState = {
+    id: number;
+    enabled: boolean;
+    description?: string;
+    created: number;
+    modified: number;
+    meta: Record<string, unknown>;
+    brand_t?: string;
+    retailer_t?: string;
+    vendor_t?: string;
+    name: string;
+    justification?: string;
+    timestamp: number;
+    email: string;
+}
+
+export type Attachment = {
+    name: string;
+    key?: string;
+    kind: string;
+    content_type?: string;
+    url?: string;
+}
+
+export type ComponentPrice = {
+    rule_id: number;
+    rule_count: number;
+    price_taxes: number;
+    vat: number;
+    price: number;
+    currency: string;
+    hs_code: number;
+    notes: string[];
+    hs_code_priority: number;
+    price_original: number;
+    vat_included: boolean;
+    ddp_included: boolean;
+    fixed_price: boolean;
+    round_price: boolean;
+    base_country: string;
+    base_currency: string;
+    exchanged?: boolean;
+    cites: boolean;
+}
+
+export type TotalPrice = {
+    price_final: number;
+    price_taxes: number;
+    vat: number;
+    ddp: number;
+    ddp_percent: number;
+    vat_rate: number;
+    price: number;
+    currency: string;
+    country: string;
+    hs_codes: number[];
+    hs_code_priority: number;
+    shipping:number;
+    notes: string[];
+    diag: {
+        no_round: {
+            price_final: number;
+            price_taxes: number;
+            price: number;
+            vat: number;
+            ddp: number;
+        };
+        patch: boolean;
+        round_value: boolean;
+        is_vat: boolean;
+        is_ddp: boolean;
+        is_european: boolean;
+        is_rounded: boolean;
+        vat_included: boolean;
+        ddp_included: boolean;
+        fixed_price: boolean;
+        round_price: boolean;
+        base_country: string;
+        base_currency: string;
+        exchanged: boolean;
+    },
+    cites: boolean;
+}
+
+export type OrderDetails = {
+    brand: string;
+    model: string;
+    variant?: string;
+    parts: Part[];
+    parts_m: Record<string, {
+        material: string;
+        color: string;
+    }>;
+    gender: string;
+    size: number;
+    query: string;
+    url: string;
+    image: string;
+    extras: number;
+}
+
+export enum OrderStatus {
+    pending = "pending",
+    created = "created",
+    production = "production",
+    ready = "ready",
+    sent = "sent",
+    received = "received",
+    returned = "returned",
+    canceled = "canceled"
+}
+
+export enum OrderProductionStatus {
+    unset = "unset",
+    waiting = "waiting",
+    cutting = "cutting",
+    sewing = "sewing",
+    assembling = "assembling",
+    produced = "produced",
+    canceled = "canceled"
+}
+
+export enum Gender {
+    male = "male",
+    female = "female",
+    kids = "kids",
+    unknown = "unknown"
+}
+
+export type Order = {
```

Comment:
Try ordering all these types from "outer" (Order) to "inner" (ComponentPrice/TotalPrice)